### PR TITLE
Add lintspaces to enforce space-related rules #7433

### LINKED ISSUE #7433: Add lintspaces to enforce space-related rules
[lintspaces](https://github.com/evanshortiss/lintspaces-cli) can enforce some rules such as:
- No trailing whitespace
- No tab character
- Each file ending with EOL character
- No multiple empty lines (can even be configured to set the max number of empty lines)

While the existing linters like CheckStyle, ESLint, and stylelint can enforce those rules for their respective languages, lintspaces is language-agnostic and can lint files with practically any extension. We can consider using it for HTML, JSP, tag, JSON, and XML files if not more.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #7433 

**Outline of Solution**

Added lintspaces and enforced the said four rules, including fixing all violations (all fixes are automated with Eclipse's search-and-replace).

While adding lintspaces to the list of linters should still be discussed, at least this PR still brings the benefit of fixing all the found violations.

To reviewer(s): the number of LoCs changed may look scary but there is nothing really worth reviewing (a `?w=1` will confirm that).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +26,7 @@
     "lint:test": "eslint src/main/webapp/dev -c static-analysis/teammates-eslint.yml --ext .es6",
     "lint:json": "eslint src/main/webapp/js/*.json src/main/resources/*.json src/test/resources/data/*.json -c static-analysis/teammates-eslint-json.yml",
     "lint:css": "stylelint src/main/webapp/stylesheets/*.css --config static-analysis/teammates-stylelint.yml",
-    "lint": "npm run lint:src && npm run lint:test && npm run lint:json && npm run lint:css"
+    "lint:spaces": "lintspaces -n -t -d spaces -l 1 'src/**/*.jsp' 'src/**/*.tag' 'src/main/**/*.html' 'src/**/*.xml' 'src/**/*.json' 'src/**/*.css' 'src/**/*.java' 'src/**/*.es6'",
```
Comment (whipermr5) [inline package.json]: Can we include `.properties`, `.yml` and `.gradle` files too?

**Comment by whipermr5** [comment]
@wkurniawan07 Thanks so much for contributing code! This PR makes me really happy 😄💯

**Comment by wkurniawan07** [comment]
Not sure if it's worth the enforcement, but I extended the lint to `.properties`, `.yml`, and `.gradle` files as well. Also intended to extend to `.md` files, but I couldn't for the reason that changes confined to `.md` files skips CI. Nevertheless I took this chance to fix violations in `.md` files.

**Comment by wkurniawan07** [comment]
@whipermr5 I need to disagree on many part of this.

- Moving the git doc-only check to after `npm run lint` will run all the linting routines as we know it for doc-only changes. It's easily 5 minutes worth of effort wasted.
- If we really want to go markdown-linting, I'd very much prefer [a full-blown markdown linter](https://github.com/DavidAnson/markdownlint).
- I don't think the position of the git doc-only check affects the Firefox installation because if it's indeed the case all docs-only PR should skip it even with the current configuration, but they did not. Besides, Firefox installation takes only few seconds.

**Comment by whipermr5** [comment]
@wkurniawan07 Okay then; agree we should keep the doc-only build as short as possible, so the current change set is fine. (Btw, by "firefox installation stuff" I meant [these lines](https://github.com/TEAMMATES/teammates/blob/8d041e8e22cc62ad7306710e3faea4d1c8a09062/.travis.yml#L33-L35), as it wouldn't have made sense to have them in the doc-only build.)

**Pull request merged**